ChopChop/Views/Ingredients/IngredientCollectionView.swift

```diff
@@ -0,0 +1,180 @@
+import SwiftUI
+
+struct IngredientCollectionView: View {
+    @EnvironmentObject var settings: UserSettings
+    @ObservedObject var viewModel: IngredientCollectionViewModel
+
+    let columns = [
+        GridItem(),
+        GridItem(),
+        GridItem(),
+        GridItem()
+    ]
+
+    var body: some View {
+        VStack {
+            SearchBar(text: $viewModel.query, placeholder: "Search ingredients...")
+            HStack {
+                Spacer()
+                Button(action: {
+                    withAnimation {
+                        viewModel.filterByExpiryDate.toggle()
+                    }
+                }) {
+                    Text("Filter by expiry date")
+                }
+            }
+            .padding([.leading, .trailing])
+
+            if viewModel.filterByExpiryDate {
+                ExpiryDatePicker()
+            }
+
+            if viewModel.ingredients.isEmpty {
+                NotFoundView()
+            } else {
+                switch settings.viewType {
+                case .list:
+                    ListView()
+                case .grid:
+                    GridView()
+                }
+            }
+        }
+        .navigationTitle(Text(viewModel.title))
+        .toolbar {
+            HStack {
+                Text("View type:")
+                Picker("View by", selection: $settings.viewType) {
+                    Text("List").tag(UserSettings.ViewType.list)
+                    Text("Grid").tag(UserSettings.ViewType.grid)
+                }
+            }
+        }
+        .onDisappear {
+            viewModel.query = ""
+            viewModel.filterByExpiryDate = false
+            viewModel.expiryDateStart = .today
+            viewModel.expiryDateEnd = .today
+        }
+    }
+
+    func ExpiryDatePicker() -> some View {
```

Comment:
Any reason why this and the subsequent views with no arguments are functions and not computed properties? similar to `body` like
```
var expiryDatePicker: some View {}
```

Reply:
A good point, I'll change them.